src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -318,19 +345,33 @@ def __init__(self, x, y, curve):
     def public_key(self, backend):
         return backend.load_elliptic_curve_public_numbers(self)
 
-    def encode_point(self):
+    def encode_point(self, compressed=False):
         # key_size is in bits. Convert to bytes and round up
         byte_length = (self.curve.key_size + 7) // 8
-        return (
-            b'\x04' + utils.int_to_bytes(self.x, byte_length) +
-            utils.int_to_bytes(self.y, byte_length)
-        )
+        if not compressed:
+            return (
+                b'\x04' + utils.int_to_bytes(self.x, byte_length) +
+                utils.int_to_bytes(self.y, byte_length)
+            )
+        else:
+            compression_byte = b'\x03' if self.y % 2 else b'\x02'
+            return (
+                compression_byte + utils.int_to_bytes(self.x, byte_length)
+            )
 
     @classmethod
     def from_encoded_point(cls, curve, data):
         if not isinstance(curve, EllipticCurve):
             raise TypeError("curve must be an EllipticCurve instance")
 
+        warnings.warn(
```

Comment:
Could you add a `with pytest.warns(CryptographyDeprecationWarning):` to the tests for this function so we can be sure it's raising it (and also so we document that it's deprecated in the tests as well)